stablehlo/dialect/TypeInference.h

```diff
@@ -144,6 +144,13 @@ LogicalResult inferSortOp(
     Region& comparator,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes);
 
+LogicalResult inferTransposeOp(MLIRContext* context, Optional<Location> loc,
+                               Value operand,
+                               llvm::SmallVector<int64_t>& inputBounds,
+                               DenseIntElementsAttr permutation,
+                               SmallVectorImpl<Type>& inferredReturnTypes,
+                               llvm::SmallVector<int64_t>& resultBounds);
```

Comment:
I think we decided to use `inferReturnTypes` in the discussions back then because we thought that `inferReturnTypeComponents` doesn't support encodings. If it does, then let's use `inferReturnTypeComponents` for consistency with everything else.

Reply:
This sounds good to me.

Reply:
Ah `inferReturnTypeComponents` does not work: this assert block it:
[inferReturnTypeComponents](https://github.com/llvm/llvm-project/blob/main/mlir/lib/Interfaces/InferTypeOpInterface.cpp#L190)